src/NUnitFramework/framework/Internal/Extensions/TypeExtensions.cs

```diff
@@ -8,5 +8,20 @@ internal static class TypeExtensions
     {
         public static bool ImplementsIComparable(this Type type) =>
             type?.GetInterface("System.IComparable") != null;
+
+        public static bool IsSortable(this Type type)
+        {
+            if (!type.ImplementsIComparable())
+                return false;
+
+            if (TypeHelper.IsTuple(type) || TypeHelper.IsValueTuple(type))
+            {
+                foreach (var typeArg in type.GetGenericArguments())
+                    if (!typeArg.IsSortable())
```

Comment:
Recursively check + confirm that all of a tuple's properties can reasonably be expected to be sortable.